### ISSUE TITLE
zappa update errors in docker container with AWS profile/env vars

### ISSUE THREAD

**Issue by SmileSydney** [Miserlou/Zappa#1934]
`zappa update dev `in docker container errors:
`Warning! Couldn't get function prop-scrape-serverless-dev in ap-southeast-2 - have you deployed yet?`

I get this error in either of the two scenrios:
1. Set AWS_XXXX env vars and removed "profile_name" from zappa_settings.json.
2. Defined "default" AWS profile and set "profile_name": "default"

Yes virtual on 3.7

## Expected Behavior

Should update AWS lambda function

## Actual Behavior

`Warning! Couldn't get function prop-scrape-serverless-dev in ap-southeast-2 - have you deployed yet?`

## Possible Fix

Verbose logging would help to understand what the actual problem is

## Steps to Reproduce

0. Set environment vars on docker host (passed in in next step)
1. docker run --rm -it -e AWS_DEFAULT_REGION="%AWS_DEFAULT_REGION%" -e AWS_ACCESS_KEY_ID="%AWS_ACCESS_KEY_ID%" -e AWS_SECRET_ACCESS_KEY="%AWS_SECRET_ACCESS_KEY%" -v "%CD%":/var/task lambci/lambda:build-python3.7 bash
2. python -m venv aws_venv
3. source aws_venv/bin/activate
4. pip install -r aws_requirements.txt
5. zappa update

Get error 

## Your Environment

* Zappa version used: 0.48.2
* Operating System and Python version: Docker host: Windows 10
Docker container: lambci/lambda:build-python3.7
* The output of `pip freeze`:
argcomplete==1.9.3
boto3==1.9.233
botocore==1.12.233
certifi==2019.9.11
cfn-flip==1.2.1
chardet==3.0.4
Click==7.0
docutils==0.14
durationpy==0.5
future==0.16.0
hjson==3.0.1
idna==2.8
jmespath==0.9.3
kappa==0.6.0
lambda-packages==0.20.0
placebo==0.9.0
python-dateutil==2.7.5
python-slugify==1.2.4
PyYAML==5.1.2
requests==2.22.0
s3transfer==0.2.1
selenium==3.141.0
six==1.11.0
SQLAlchemy==1.3.8
toml==0.10.0
tqdm==4.19.1
troposphere==2.5.1
Unidecode==1.1.1
urllib3==1.24.1
Werkzeug==0.16.0
wsgi-request-logger==0.4.6
zappa==0.48.2

* Link to your project (optional):
* Your `zappa_settings.py`: 
{
  "dev": {
    "aws_region": "ap-southeast-2",
    "project_name": "prop-scrape-serverless",
    "use_apigateway": false,
    "runtime": "python3.7",
    "s3_bucket": "gaia-zappa-deploy",
    "keep_warm": false,
    "timeout_seconds": 120,
    "memory_size": 1024,
    "environment_variables": {
      "PATH": "/var/task/chrome"
    },
    "events": [
      {
        "function": "prop_search_scope.lambda_handler",
        "event_source": {
          "arn": "arn:aws:sqs:ap-southeast-2:XXXXXXX:YYYYYYYY",
          "batch_size": 1,
          "enabled": true
        }
      },
      {
        "function": "prop_search_page.lambda_handler",
        "event_source": {
          "arn": "arn:aws:sqs:ap-southeast-2:XXXXXXX:YYYYYYYY",
          "batch_size": 1,
          "enabled": true
        }
      },
      {
        "function": "prop_listing_page.lambda_handler",
        "event_source": {
          "arn": "arn:aws:sqs:ap-southeast-2:XXXXXXX:YYYYYYYY",
          "batch_size": 1,
          "enabled": true
        }
      }
    ]
  }
}

**Comment by SmileSydney** [comment]
#1928

**Comment by jcomish** [comment]
Have you figured anything out about this problem? I have been running into the same problem with no progress... When I try to update from my CI build step, it injects the directory name before the environment for some reason. So...

Files are in /tmp/src
zappa update test from /tmp/src

I get the following error message:
Couldn't get function src-test in us-east-1 - have you deployed yet?

**Comment by jneves** [comment]
If you run "aws sts get-caller-identity" do you get the user you expect?

**Comment by jcomish** [comment]
Yes, I get the user that I expect, and the region is in us-east-1. For some reason it is prepending src- to the environment name. I even replaced it with "zappa deploy test" to see what happens, and it deploys "src-test" which I know shouldn't be possible...

Not sure if this is the same error that is being reported here. This is all being done in AWS CodeBuild.

**Comment by jcomish** [comment]
Ah I see the problem, zappa appends the directory name to make it unique... of course!
